### DISCUSSION TITLE
axis scaling in 2D line plot

### DISCUSSION THREAD

**Discussion by c-humbel** [alandefreitas/matplotplusplus#132] [Q&A] [answered]
Hi,
A very simple question, but I didn't find an answer in the documentation: Is there a way to get equal scaling of the x- and y-axis in a line plot?
For a student project, i had to plot sine vs. cosine, and I would like the plot to look like a circle and not an ellipse.
[circle.pdf](https://github.com/alandefreitas/matplotplusplus/files/6242249/circle.pdf)
Matplotlib has a function ```set_aspect()``` that achieves this, is there something similar for matplot++?
The code I used is very simple, but for completion I post it anyway:
```c++
double t = 0;
double dt = 2*M_PI / size;
for (unsigned i = 0; i < size; ++i) {
    x[i] = std::sin(t);
    y[i] = std::cos(t);
    t += dt;
}
matplot::plot(x, y);
matplot::save("circle.pdf");
```

**Comment by alandefreitas** [comment, marked as answer]
Maybe one of these functions can help you: https://github.com/alandefreitas/matplotplusplus/blob/7fd8c9ecfac50f20ca1b8e92803c8b36e1af56f6/source/matplot/core/axes_type.h#L106

**Reply by c-humbel** [reply]
Thanks a lot, this worked!
